agent/engine/execcmd/agent_version.go

```diff
@@ -0,0 +1,64 @@
+package execcmd
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+const (
+	versionDelimiter  = "."
+	badInputReturn    = -999
+	badFormatErrorStr = "invalid version format"
+)
+
+// compareAgentVersion compares two versions, returning -1 if v1 < v2, 0 if v1 == v2, and 1 if v1 > v2.
+// The version must have only integers representing major, minor, patch, and reserved; delimited by dots (e.g. "1.2.3.4").
+// An error is returned if any of the versions passed as parameter is malformed.
+// example:
+// SemverCompare("3.0.236.0", "3.0.237.0") returns -1, nil
+// SemverCompare("3.0.236.0", "3.0.236.0") returns 0, nil
+// SemverCompare("3.0.237.0", "3.0.236.0") returns 1, nil
+// SemverCompare("a.b.c.d", "3.0.236.0") returns -999, error
+func compareAgentVersion(v1Str, v2Str string) (int, error) {
+	v1Arr := strings.Split(v1Str, versionDelimiter)
+	v2Arr := strings.Split(v2Str, versionDelimiter)
+
+	if len(v1Arr) != len(v2Arr) {
+		return badInputReturn, errors.New(badFormatErrorStr)
+	}
+	for i := 0; i < len(v1Arr); i++ {
+		v1Int, err := strconv.Atoi(v1Arr[i])
```

Comment:
I'm wondering if this should be `https://golang.org/pkg/strconv/#ParseUint` instead

Reply:
great suggestion! changed and added checks for negative versions :)